test/integration/v1/constants.py

```diff
@@ -1,10 +1,154 @@
+  strategy: Literal["constraint"]
+
+
+BoundType = TypeVar("BoundType", bound=int | float)
+
+
+class BoundsMetaType(TypedDict, Generic[BoundType]):
+  min: BoundType
+  max: BoundType
+
+
+class ParameterMetaType(TypedDict):
+  name: str
+  conditions: NotRequired[dict[str, str | list[str]] | None]
+
+
+class NumericalParameterMetaType(ParameterMetaType):
+  transformation: NotRequired[Literal["log"] | Literal["none"] | None]
+
+
+class IntegerParameterMetaType(NumericalParameterMetaType):
+  type: Literal["int"]
+
+
+class BoundedIntegerParameterMetaType(IntegerParameterMetaType):
+  bounds: BoundsMetaType[int]
+
+
+class GriddedIntegerParameterMetaType(IntegerParameterMetaType):
+  grid: list[int]
+
+
+class ParameterPriorMetaType(TypedDict):
+  name: str
+
+
+class NormalParameterPriorMetaType(ParameterPriorMetaType):
+  mean: float
+  scale: float
+
+
+class BetaParameterPriorMetaType(ParameterPriorMetaType):
+  shape_a: float
+  shape_b: float
+
+
+class DoubleParameterMetaType(NumericalParameterMetaType):
+  type: Literal["double"]
+  prior: NotRequired[NormalParameterPriorMetaType | BetaParameterPriorMetaType | None]
+
+
+class BoundedDoubleParameterMetaType(DoubleParameterMetaType):
+  bounds: BoundsMetaType[float]
 
 
-DEFAULT_EXPERIMENT_META = dict(
+class GriddedDoubleParameterMetaType(DoubleParameterMetaType):
+  grid: list[float]
+
+
+class CategoricalValueMetaType(TypedDict):
+  name: str
+
+
+class CategoricalParameterMetaType(ParameterMetaType):
+  type: Literal["categorical"]
+  categorical_values: list[CategoricalValueMetaType | str]
+
+
+AnyParameterMetaType = (
+  BoundedIntegerParameterMetaType
+  | GriddedIntegerParameterMetaType
+  | BoundedDoubleParameterMetaType
+  | GriddedDoubleParameterMetaType
+  | CategoricalParameterMetaType
+)
+
+
+class ConditionalMeta(TypedDict):
+  name: str
+  values: list[str]
+
+
+class LinearConstraintTermMeta(TypedDict):
+  name: str
+  weight: float
+
+
+class LinearConstraintMeta(TypedDict):
+  type: Literal["greater_than"] | Literal["less_than"]
+  terms: list[LinearConstraintTermMeta]
+  threshold: float
+
+
+class TaskMeta(TypedDict):
+  name: str
+  cost: float
+
+
+class BaseExperimentMetaType(TypedDict):
+  type: NotRequired[str]
+  name: NotRequired[str | None]
+  parameters: list[AnyParameterMetaType]
+  conditionals: NotRequired[list[ConditionalMeta] | None]
+  num_solutions: NotRequired[int | None]
+  linear_constraints: NotRequired[list[LinearConstraintMeta] | None]
+
+
+AnyMetricMetaType = StoredMetricMetaType | OptimizedMetricMetaType | ConstrainedMetricMetaType
+
+
+class CoreExperimentMetaType(BaseExperimentMetaType):
+  metric: NotRequired[AnyMetricMetaType | None]
+  metrics: NotRequired[list[AnyMetricMetaType] | None]
+  observation_budget: NotRequired[int | None]
+  project: NotRequired[str | None]
+  tasks: NotRequired[list[TaskMeta]]
+
+
+class AiExperimentMetaType(BaseExperimentMetaType):
+  metrics: list[AnyMetricMetaType]
+  budget: NotRequired[int | None]
```

Comment:
This was pretty experimental, but it could be a great way to help validate inputs in the python client.